pkg/sql/sqlbase/index_encoding.go

```diff
@@ -270,26 +271,17 @@ func MakeFullKeyFromEncDatums(
 		return nil, err
 	}
 
-	primaryIndex := tableDesc.PrimaryIndex
-	extraColumns := index.ExtraColumnIDs
-	// PrimaryImplicitIdxs maps the position of the implicit column in the key
-	// to the appropriate column in the primary index.
-	primaryImplicitIdxs := make([]int, len(extraColumns))
-	for i, id := range extraColumns {
-		for j, primaryID := range primaryIndex.ColumnIDs {
-			if id == primaryID {
-				primaryImplicitIdxs[i] = j
-			}
-		}
-	}
-	implicitDirs := make([]IndexDescriptor_Direction, len(primaryImplicitIdxs))
-	for i, idx := range primaryImplicitIdxs {
-		implicitDirs[i] = primaryIndex.ColumnDirections[idx]
-	}
-	implicitTypes := make([]ColumnType, len(extraColumns))
-	for i, id := range extraColumns {
-		implicitTypes[i] = tableDesc.ColumnTypes()[id]
+	implicitTypes := make([]ColumnType, len(implicitValues))
```

Comment:
This seems like something we should be passing in as an argument, rather than reconstructing every time. It's static for the whole lifetime on the lookup join, isn't it? The lookup join should have this slice created at plan time and pass it in to the key creation function, rather than having to do this allocation every time.